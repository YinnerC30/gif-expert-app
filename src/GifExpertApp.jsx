import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["Goku", "Vegeta"]);
  console.log(categories);

  const onAddCategory = () => {
    setCategories([...categories, "Popeye"]);
  };
  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory />
      <button type="button" onClick={onAddCategory}>
        Agregar
      </button>

      <ol>
        {categories.map((category) => {
          return <li key={category}>{category}</li>;
        })}
      </ol>
    </>
  );
};
