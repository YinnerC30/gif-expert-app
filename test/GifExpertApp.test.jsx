import { render, screen } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe("Pruebas a <GifExpertApp/>", () => {
  test("Mostrar contenido renderizado", () => {
    render(<GifExpertApp />);
    screen.debug();
  });

  test("Toma de snapshot", () => {
    const { container } = render(<GifExpertApp />);
    expect(container).toMatchSnapshot();
  });

  test("Debe contener el titulo de la aplicación", () => {
    render(<GifExpertApp />);
    expect(screen.getByText("GifExpertApp")).toBeTruthy();
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      "GifExpertApp"
    );
  });

  test("Goku debe ser la categoria de Gif predeterminada al ejecutar la aplicación", () => {
    render(<GifExpertApp />);
    expect(screen.getByText("Goku")).toBeTruthy();
  });

  test("Se debe mostrar el mensaje cargando...", () => {
    render(<GifExpertApp />);
    expect(screen.getByText("Cargando...")).toBeTruthy();
  });
});
