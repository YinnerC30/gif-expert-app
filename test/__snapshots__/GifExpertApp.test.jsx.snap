// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a <GifExpertApp/> Toma de snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    action=""
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    Goku
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
